File: backend/app/schemas/local_data.py
from typing import Literal

from sqlmodel import SQLModel


class LocalDataIssueCounts(SQLModel):
    error: int
    warning: int
    info: int


class LocalDataIssueSummary(SQLModel):
    total: int
    by_severity: LocalDataIssueCounts


class LocalDataIssueRead(SQLModel):
    severity: Literal["error", "warning", "info"]
    code: str
    target: str
    message: str


class LocalDataMediaSummary(SQLModel):
    records: int
    approved: int
    pending: int
    rejected: int
    unknown_status: int


class LocalDataPlaceSummary(SQLModel):
    records: int
    published: int
    draft: int
    archived: int
    unknown_status: int


class LocalDataPublicPayloadSummary(SQLModel):
    checked: int


class LocalDataStorageSummary(SQLModel):
    private_files: int
    public_files: int
    orphan_private_files: int
    orphan_public_files: int
    private_bytes: int
    public_bytes: int


class LocalDataDiagnosticsSummary(SQLModel):
    photos: LocalDataMediaSummary
    memories: LocalDataMediaSummary
    places: LocalDataPlaceSummary
    public_payloads: LocalDataPublicPayloadSummary
    storage: LocalDataStorageSummary
    issues: LocalDataIssueSummary


class LocalDataDiagnosticsRead(SQLModel):
    generated_at: str
    status: Literal["ok", "warning", "error"]
    summary: LocalDataDiagnosticsSummary
    issues: list[LocalDataIssueRead]


class LocalDataCleanupActionRead(SQLModel):
    action: Literal["delete_file"]
    applied: bool
    relative_path: str
    storage: Literal["private", "public"]
    status: Literal["planned", "deleted", "missing", "not-file"]


class LocalDataCleanupRead(SQLModel):
    mode: Literal["apply", "dry-run"]
    status: Literal["ok", "warning", "error"]
    actions: list[LocalDataCleanupActionRead]
    diagnostics: LocalDataDiagnosticsRead
    diagnostics_before: LocalDataDiagnosticsRead | None

File: frontend/src/api/types/localData.ts
type LocalDataStatus = "ok" | "warning" | "error";
type LocalDataIssueSeverity = "error" | "warning" | "info";

type LocalDataIssueCounts = {
  error: number;
  warning: number;
  info: number;
};

type LocalDataIssueSummary = {
  total: number;
  by_severity: LocalDataIssueCounts;
};

type LocalDataIssue = {
  severity: LocalDataIssueSeverity;
  code: string;
  target: string;
  message: string;
};

type LocalDataMediaSummary = {
  records: number;
  approved: number;
  pending: number;
  rejected: number;
  unknown_status: number;
};

type LocalDataPlaceSummary = {
  records: number;
  published: number;
  draft: number;
  archived: number;
  unknown_status: number;
};

type LocalDataStorageSummary = {
  private_files: number;
  public_files: number;
  orphan_private_files: number;
  orphan_public_files: number;
  private_bytes: number;
  public_bytes: number;
};

export type LocalDataDiagnostics = {
  generated_at: string;
  status: LocalDataStatus;
  summary: {
    photos: LocalDataMediaSummary;
    memories: LocalDataMediaSummary;
    places: LocalDataPlaceSummary;
    public_payloads: {
      checked: number;
    };
    storage: LocalDataStorageSummary;
    issues: LocalDataIssueSummary;
  };
  issues: LocalDataIssue[];
};

export type LocalDataCleanupReport = {
  mode: "apply" | "dry-run";
  status: LocalDataStatus;
  actions: Array<{
    action: "delete_file";
    applied: boolean;
    relative_path: string;
    storage: "private" | "public";
    status: "planned" | "deleted" | "missing" | "not-file";
  }>;
  diagnostics: LocalDataDiagnostics;
  diagnostics_before: LocalDataDiagnostics | null;
};
